From 3eedd7115bebe064845bfc732c3498511511c434 Mon Sep 17 00:00:00 2001
From: Tony LIU <junjie.liu@freescale.com>
Date: Fri, 20 Jul 2012 10:11:06 +0800
Subject: [PATCH] ENGR00217717 mfgtool firmware will crash during mfgtool
 running

- the root cause of this issue is there is no protection for
  the resource which will be accessed by multiple thread

Signed-off-by: Tony LIU <junjie.liu@freescale.com>
---
 drivers/usb/gadget/fsl_updater.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/gadget/fsl_updater.c b/drivers/usb/gadget/fsl_updater.c
index 8b2b535..32415d5 100644
--- a/drivers/usb/gadget/fsl_updater.c
+++ b/drivers/usb/gadget/fsl_updater.c
@@ -70,9 +70,11 @@ static u32 count_list(struct list_head *l)
 	u32 count = 0;
 	struct list_head *tmp;
 
+	mutex_lock(&utp_context.lock);
 	list_for_each(tmp, l) {
 		count++;
 	}
+	mutex_unlock(&utp_context.lock);
 
 	return count;
 }
-- 
1.8.0

